paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsActivity.kt

```diff
@@ -46,13 +51,29 @@ internal class PaymentOptionsActivity : BaseSheetActivity<PaymentOptionResult>()
 
         setContentView(viewBinding.root)
 
-        viewModel.paymentOptionResult.launchAndCollectIn(this) {
-            closeSheet(it)
-        }
-
         viewBinding.content.setContent {
             StripeTheme {
-                PaymentOptionsScreen(viewModel)
+                val isProcessing by viewModel.processing.collectAsState()
+                val result by viewModel.paymentOptionResult.collectAsState(initial = null)
+
+                val bottomSheetState = rememberBottomSheetState(
+                    confirmValueChange = { !isProcessing },
```

Comment:
The current implementation of `ModalBottomSheetLayout` doesn’t allow us to fully disable swipe gestures (we’d need Compose 1.5.0 for that; see [here](https://github.com/stripe/stripe-android/pull/7142)). However, we do want to disable dismissal while we’re processing, which is what `confirmValueChange` does.